src/migrateCalculatedMeasures.ts

```diff
@@ -0,0 +1,89 @@
+import { ContentRecord } from "@activeviam/activeui-sdk";
+import { getCalculatedMeasures } from "./getCalculatedMeasures";
+import { generateId } from "./generateId";
+
+export interface LegacyCalculatedMeasure {
+  expression: string;
+  formatStringExpression: string;
+  uniqueName: string;
+  owners: string[];
+  readers: string[];
+  isDirectory: boolean;
+}
+
+/**
+ * Returns the converted content record tree for the calculted measures, ready to be used in ActiveUI 5.
+ */
+export async function migrateCalculatedMeasures(
+  legacyPivotFolder: ContentRecord
+): Promise<ContentRecord> {
+  const entitlements = legacyPivotFolder?.children?.entitlements;
+  const calculatedMeasuresFolder = entitlements?.children?.cm!;
+
+  const legacyCalculatedMeasures = calculatedMeasuresFolder
+    ? await getCalculatedMeasures(calculatedMeasuresFolder)
+    : [];
+
+  const contentFolder = {
+    entry: { isDirectory: true, owners: ["ROLE_USER"], readers: ["ROLE_USER"] },
+    children: {},
+  };
+
+  const structureFolder = {
+    entry: { isDirectory: true, owners: ["ROLE_USER"], readers: ["ROLE_USER"] },
+    children: {},
+  };
+
+  legacyCalculatedMeasures.forEach((legacyMeasure) => {
+    const { owners, readers, expression, formatStringExpression, uniqueName } =
+      legacyMeasure;
+
+    const id = generateId();
+
+    contentFolder.children = {
+      ...contentFolder.children,
+      [id]: {
+        entry: {
+          owners,
+          readers,
+          isDirectory: false,
+          content: JSON.stringify({
+            expression,
+            properties: [`FORMAT_STRING=${formatStringExpression}`],
```

Comment:
In the MDX spec, calculated members can have other properties than `FORMAT_STRING` (for example `MEMBER_CAPTION`).
Could you check what happens in AUI4 if you save a calculated measure with such a property ?
If it ends up in the XML, we should take it into account when migrating.

Reply:
@antoinetissier I've just checked AUI-4, and it looks like when creating a new calculated measure, we support `FORMAT_STRING` and custom styling. I don't see how a user can save a calculated measure with the property of `MEMBER_CAPTION`. Additionally, it looks like when you save a measure in AUI-5, the only modification that can be done is to `FORMAT_STRING`, so we don't even support additional properties at the moment. Should I leave this as is? What do you think?

Reply:
I was thinking of trying to add the property manually in the query editor in ActiveUI 4, saving the calculated measure, and see if the property ends up on the content server.
But thinking more about it, I think the measures saved this way were actually published to the cube, and so that they were not inlined in the MDX WHERE clause.
In this case it's all good 👍